### PR TITLE
fix 22189

### PR DESCRIPTION
**Pull request by smichr**
#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by smichr** [comment]
@oscargus , just trying a different approach

File: sympy/core/expr.py [hunk 1]
```diff
@@ -2479,6 +2485,12 @@ def could_extract_minus_sign(self):
         >>> {i.could_extract_minus_sign() for i in (e, -e)}
         {False, True}
 
+        TO BE REMOVED
+        >>> from sympy import signsimp, sqrt
```
Comment (smichr) [inline sympy/core/expr.py:2489]:
```suggestion
        >>> from sympy import sqrt
```

File: sympy/core/expr.py [hunk 2]
```diff
@@ -2284,15 +2284,21 @@ def extract_multiplicatively(self, c):
...
-                    else:
-                        xc = cs.extract_multiplicatively(c)
-                    if xc is not None:
-                        return xc*ps  # rely on 2-arg Mul to restore Add
+            if c.is_Number:
+                if c is S.NegativeOne:
+                    from sympy.core.function import _coeff_isneg
+                    neg_args = sum(1 for _ in self.args if _coeff_isneg(_))
+                    xs = 2*neg_args - len(self.args)  # excess of negatives if positive
+                    return xs > 0 if xs else bool(self.sort_key() < (-self).sort_key())
```
Comment (smichr) [inline sympy/core/expr.py:2292]:
```suggestion
                   neg = (xs > 0) if xs else bool(self.sort_key() < (-self).sort_key())
                   if neg:
                       return -self
```

File: sympy/core/expr.py [hunk 3]
```diff
@@ -2284,15 +2284,23 @@ def extract_multiplicatively(self, c):
...
-                    if xc is not None:
-                        return xc*ps  # rely on 2-arg Mul to restore Add
+            if c.is_Number:
+                if c is S.NegativeOne:
+                    from sympy.core.function import _coeff_isneg
+                    neg_args = sum(1 for _ in self.args if _coeff_isneg(_))
+                    xs = 2*neg_args - len(self.args)  # excess of negatives if positive
+                   neg = (xs > 0) if xs else bool(self.sort_key() < (-self).sort_key())
+                   if neg:
+                       return -self
```
Comment (smichr) [inline sympy/core/expr.py:2294]:
```suggestion
                    neg = (xs > 0) if xs else bool(self.sort_key() < (-self).sort_key())
                    if neg:
                        return -self
```

**Comment by oscargus** [comment]
As the core for #22189 is the same and this improves some other things as well, I'd say go for this!

File: sympy/simplify/simplify.py [hunk 4]
```diff
@@ -345,18 +345,14 @@ def hypersimilar(f, g, k):
 
 def signsimp(expr, evaluate=None):
     """Make all Add sub-expressions canonical wrt sign.
-
```
Comment (oscargus) [inline sympy/simplify/simplify.py:348]: I think the consensus is to have empty lines around headings.

**Pull request closed without merging**